agent/php_user_instrument.c

```diff
@@ -308,121 +298,6 @@ static void nr_php_add_custom_tracer_common(nruserfn_t* wraprec) {
-   * pair.
-   */
-
-  /*
-   * Optimize out string manipulations; don't do them if you don't have to.
-   * For instance, if funcname doesn't match, no use comparing the classname.
-   */
-
-  if (NULL == p) {
-    return false;
-  }
-  if ((NULL == func) || (ZEND_USER_FUNCTION != func->type)) {
-    return false;
-  }
-
-  if (0 != p->lineno) {
-    /*
-     * Lineno is set in the wraprec.  If lineno doesn't match, we can exit without
-     * going on to the funcname/classname pair comparison.
-     * If lineno matches, but the wraprec filename is NULL, it is inconclusive and we
-     * we must do the funcname/classname compare.
-     * If lineno matches, wraprec filename is not NULL, and it matches/doesn't match,
-     * we can exit without doing the funcname/classname compare.
-     */
-    if (p->lineno != nr_php_zend_function_lineno(func)) {
-      return false;
-    } 
-    /*
-     * lineno matched, let's check the filename
-     */
-    filename = nr_php_function_filename(func);
-
-    /*
-     * If p->filename isn't NULL, we know the comparison is accurate;
-     * otherwise, it's inconclusive even if we have a lineno because it
-     * could be a cli call or evaluated expression that has no filename.
-     */
-    if (NULL != p->filename) {
-      if (0 == nr_strcmp(p->filename, filename)) {
-        return true;
-      }
-      return false;
-    }
-  }
-
-  if (NULL == func->common.function_name) {
-    return false;
-  }
-
-  if (0 != nr_stricmp(p->funcnameLC, ZSTR_VAL(func->common.function_name))) {
-    return false;
-  }
-  if (NULL != func->common.scope && NULL != func->common.scope->name) {
-    klass = ZSTR_VAL(func->common.scope->name);
-  }
-
-  if ((0 == nr_strcmp(p->reportedclass, klass))
-      || (0 == nr_stricmp(p->classname, klass))) {
-    /*
-     * If we get here it means lineno/filename weren't initially set.
-     * Set it now so we can do the optimized compare next time.
-     * lineno/filename is usually not set if the func wasn't loaded when we
-     * created the initial wraprec and we had to use the more difficult way to
-     * set, update it with lineno/filename now.
-     */
-    if (NULL == p->filename) {
-      filename = nr_php_function_filename(func);
-      if ((NULL != filename) && (0 != nr_strcmp("-", filename))) {
-        p->filename = nr_strdup(filename);
-      }
-    }
-    if (0 == p->lineno) {
-      p->lineno = nr_php_zend_function_lineno(func);
-    }
-    return true;
-  }
-  return false;
-}
-
-nruserfn_t* nr_php_get_wraprec_by_func(zend_function* func) {
-  nruserfn_t* p = NULL;
-
-  if ((NULL == func) || (ZEND_USER_FUNCTION != func->type)) {
-    return NULL;
-  }
-
-  p = nr_wrapped_user_functions;
-
-  while (NULL != p) {
-    if (nr_php_wraprec_matches(p, func)) {
-      return p;
-    }
-    p = p->next;
-  }
-
-  return NULL;
-}
-#endif
-
 #define NR_PHP_UNKNOWN_FUNCTION_NAME "{unknown}"
 nruserfn_t* nr_php_add_custom_tracer_callable(zend_function* func TSRMLS_DC) {
```

Comment:
Should this be `const zend_function*`?

Reply:
Not necessarily. When `LOOKUP_USE_OP_ARRAY` method is used (PHP < 7.4), `func->op_array.reserved` is modified by `nr_php_op_array_set_wraprec` which is called by `nr_php_user_instrument_set` which is called by `nr_php_wrap_zend_function` which is called later in `nr_php_add_custom_tracer_callable`.